package.json

```diff
@@ -14,10 +14,10 @@
     "test:server": "mocha \"test/unit/**/*-test.js\"",
     "test:browser": "mochify \"test/unit/**/*.js\"",
     "test:features": "node scripts/cucumber.js",
-    "coverage": "scripts/cov",
-    "coveralls": "npm run coverage mocha-lcov-reporter | coveralls",
+    "test:coverage": "nyc --reporter=text-lcov npm run test:server",
+    "coveralls": "npm run test:coverage | coveralls",
     "ci:lint": "npm run lint",
-    "ci:test": "npm run test",
+    "ci:test": "npm run test && npm run coveralls",
```

Comment:
This runs the tests twice, which will take the CI twice as much to finish the build. I think we don't really need the `coveralls` or `test:coverage` commands, maybe we could do something like

```
  "test:server": "mocha \"test/unit/**/*-test.js\""
  "test:server:coverage": "nyc --reporter=text-lcov npm run test:server"
  "ci:test": "npm run test:server:coverage (collect coverage) && npm run test:browser && npm run test:features && (send coverage to coveralls)"
```

The question is how to collect coverage like this. Does nyc support something like reporting and collecting coverage at the same time?

Reply:
You are right, no need to run unit tests twice.
We need to keep in mind that the coverage is collected only from the unit (browser) tests. This can simplify a few things.

Within CI I don't see a problem to run test + collect coverage, since we are running tests anyway. I will integrate coverage collection without duplicate tests run.